src/main/java/com/auth0/net/BaseRequest.java

```diff
@@ -37,4 +40,21 @@ public T execute() throws Auth0Exception {
         return parseResponse(response);
     }
 
+    @Override
+    public CompletableFuture<T> executeAsync() {
+        okhttp3.Request request = createRequest();
+        final CompletableFuture<T> future = CompletableFuture<T>();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                future.completeExceptionally(new Auth0Exception("Failed to execute request", e));
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                future.complete(parseResponse(response));
+            }
+        });
```

Comment:
I have the feeling you want to return `future` here.

Reply:
No need. 

I just populate the future with the result and indicate to this future that the processing is finished thanks to `.complete`.

Reply:
Yes, thanks !